### PR TITLE
typer on ga schedule script

### PR DESCRIPTION
**Pull request by JulianaCP-v2**
- typer added in the ga schedule script
- `main.py `file added
- file `common.py` with service requests added

### CODE REVIEW

File: operator_toolkit/create_soc_percentile.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def operate(args):
     :param Namespace args: argparse namespace.
     """
     sites_to_operate = Operator.sites_to_operate(args.sites_to_operate, controller=args.controller,
-                                                 config=CONFIG)
+                                                        config=CONFIG)
```
Comment (samhollenbach) [inline operator_toolkit/create_soc_percentile.py:64]: Seems like a flake8 violation

**Review by samhollenbach — APPROVED** [review]

File: operator_toolkit/operation/utils.py [hunk 2]
```diff
@@ -74,22 +79,23 @@ class HttpMethods(Enum):
...
         # load local config with operational sites if not s3
         config = FileManager.load_file(OPERATIONAL_SITES_LOCAL_PATH)
         if config:
             return config
+
+        # load s3 config with operational sites
+        config = FileManager.load_file(OPERATIONAL_SITES_S3_PATH)
+        if config:
+            return config
+
```
Comment (lucasmlofaro) [inline operator_toolkit/operation/utils.py:98]: So the result of having this `load_file()` method be nullable (can return something falsy) is that you're able to play this trick where you exit early if you find the local file and only attempt the S3 load if you make it there. This works, but you've violated DRY, it's almost the exact same snippet.

Ideally, we should make one assignment and one conditional return. We can leverage the short-circuit nature of boolean expressions (lazy evaluation) to achieve the same thing as the early exit.
```suggestion
        # load config with operational sites
        config = (FileManager.load_file(OPERATIONAL_SITES_LOCAL_PATH) or
                  FileManager.load_file(OPERATIONAL_SITES_S3_PATH))
        if config:
            ...  # deal with failed config loading
        return config

```

The other thing to note is that I typically prefer to have the "happy path" return be the default and use conditionals to do error handling.

File: operator_toolkit/main.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import typer
+
+import ga_schedule
+
+app = typer.Typer()
+
+app.add_typer(ga_schedule.app, name="ga_schedule")
```
Comment (lucasmlofaro) [inline operator_toolkit/main.py:7]: I think I understand what you were talking about before now. This is the typical pattern for separating subcommands across modules?

File: operator_toolkit/create_soc_percentile.py (hunk 1 above)
Comment (JulianaCP-v2) [reply operator_toolkit/create_soc_percentile.py:64]: you're right, thanks 👌

File: operator_toolkit/ga_schedule.py [hunk 4]
```diff
@@ -1,87 +1,48 @@
...
 
+def schedule_site(site, precool_path, loadshed_path):
+    service_manager = ServiceManager(site)
+    report = {
+        "set_suc_P_gains": service_manager.set_gains(GainType.SUC_P, OperationMode.PRECOOL),
+        "set_soc_25_percentile_gains": service_manager.set_gains(GainType.SOC_25_PERCENTILE, OperationMode.PRECOOL),
+        "set_soc_50_percentile_gains": service_manager.set_gains(GainType.SOC_50_PERCENTILE, OperationMode.PRECOOL),
+        "add_precool_schedule_rule": service_manager.add_schedule_rule(precool_path, OperationMode.PRECOOL),
+        "add_loadshed_schedule_rule": service_manager.add_schedule_rule(loadshed_path, OperationMode.LOADSHED)
+    }
```
Comment (lucasmlofaro) [inline operator_toolkit/ga_schedule.py:26]: These are potentially good places to add some structure to the data. Maybe some kind of `Report` object that has some `display()` method?

**Review by lucasmlofaro — CHANGES_REQUESTED** [review]

File: operator_toolkit/create_soc_percentile.py (hunk 1 above)
Comment (lucasmlofaro) [reply operator_toolkit/create_soc_percentile.py:64]: Yeah! I was actually gonna say, @JulianaCP-v2 maybe one task you can do this sprint would be add this repo to CircleCI.

File: operator_toolkit/main.py (hunk 3 above)
Comment (JulianaCP) [reply operator_toolkit/main.py:7]: yes, exactly

File: operator_toolkit/operation/utils.py (hunk 2 above)
Comment (JulianaCP) [reply operator_toolkit/operation/utils.py:98]: mmm nice, I like it, thanks 👌

File: operator_toolkit/ga_schedule.py (hunk 4 above)
Comment (JulianaCP) [reply operator_toolkit/ga_schedule.py:26]: mm sounds nice 🤔 will try to think in a fancy way to implement something like this, thanks

**Review by lucasmlofaro — APPROVED** [review]
Looks good

**Pull request merged**